src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -581,7 +581,17 @@ def prepare_ip_adapter_image_embeds(
 
                 image_embeds.append(single_image_embeds)
         else:
-            image_embeds = ip_adapter_image_embeds
+            image_embeds = []
```

Comment:
@asomoza 

I updated the `prepare_ip_adapter_image_embeds` method now we expect the `ip_adapter_image_embeds` to be a list of the same length as the number of IP adapters; each tensor in the list should have shape `batch_size, num_images, embed_dim` (e.g. `2, 2, 1024` in our case); the `num_image_per_prompt` is always `1` for the `ip_adapter_image_embeds` user passed, we duplicate it inside `prepare_ip_adapter_image_embeds` method; however we expected the embedding has CFG applied if we are going to use CFG in the inference (e.g. both our comfyui and diffusers style embedding has CFG applied, however, the order is different in diffusers vs comfy UI so we had to swap); 

might be a little bit confusing but it is easier to understand if you take a look at the testing script here
https://github.com/huggingface/diffusers/pull/7016#issue-2141204736

Reply:
I tested it and it works perfect for the diffusers and comfyui embeddings, also tested it with cfg and no cfg. with num_per_images > 0 and 1, it's all good, thank you for you hard work.

Reply:
How was "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/comfyui_style_test.ipadpt" obtained?

Reply:
with the `Save IPAdapter Embeds` node:

![image](https://github.com/huggingface/diffusers/assets/5442875/0cd7afca-fe46-4925-b4cc-e2cf50ec6666)